src/scripts/quest/subquest/gridania/SubFst000.cpp

```diff
@@ -0,0 +1,216 @@
+      Scene00001( player );
+    else if ( actor == Actor1 && player.getQuestSeq( getId() ) == SeqFinish )
+      Scene00004( player );
+    else if ( actor == Eobject0 )
+      Scene00002( player );
+    else if ( actor == Eobject1 )
+      Scene00003( player );
+  }
+
+  void onEventItem( Entity::Player& player, uint32_t eventId,
+                    uint32_t eventItemId, uint64_t actorId ) override
+  {
+    auto& eventMgr = Common::Service< World::Manager::EventMgr >::ref();
+    auto actor = eventMgr.mapEventActorToRealActor( static_cast< uint32_t >( actorId ) );
+
+    player.sendDebug( "onEventItem called for this quest: eventId={0}, eventItemId={1}, actor={2}", eventId, eventItemId, actor );
+
+    if ( actor == Eobject0 )
+      Scene00100( player );
+    else if ( actor == Eobject1 )
+      Scene00098( player );
+  }
+
+  private:
+
+  void checkQuestCompletion( Entity::Player& player )
+  {
+    auto currentUsed  = player.getQuestUI8AL( getId() );
+    auto currentItems = player.getQuestUI8BH( getId() );
+
+    player.sendQuestMessage( getId(), 1, 2, currentUsed + 1, 2 );
+
+    if( currentUsed + 1 >= 2 )
+    {
+      player.updateQuest( getId(), SeqFinish );
+    }
+    else
+    {
+      player.setQuestUI8AL( getId(), currentUsed  + 1 );
+      player.setQuestUI8BH( getId(), currentItems - 1 );
+    }
+  }
+
+  //////////////////////////////////////////////////////////////////////
+  // Available Scenes in this quest, not necessarly all are used
+  void Scene00000( Entity::Player& player )
+  {
+    auto callback = [ & ]( Entity::Player& player, const Event::SceneResult& result )
+    {
+      if ( result.param2 == 1 ) // accept quest
+      {
+        player.updateQuest( getId(), Seq1 );
+      }
+    };
+
+    player.playScene( getId(), 0, HIDE_HOTBAR, callback );
+  }
+
+  void Scene00001( Entity::Player& player )
+  {
+    auto callback = [ & ]( Entity::Player& player, const Event::SceneResult& result )
+    {
+      player.setQuestUI8AL( getId(), 0 ); // number of censers used
+      player.setQuestUI8BH( getId(), 2 ); // Azeyma Rose Oil key item
+      player.sendQuestMessage( getId(), 0, 0, 0, 0 );
+      player.updateQuest( getId(), Seq2 );
+    };
+
+    player.playScene( getId(), 1, HIDE_HOTBAR, callback );
+  }
+
+  void Scene00002( Entity::Player& player )
+  {
+    auto callback = [ & ]( Entity::Player& player, const Event::SceneResult& result )
+    {
+    };
+
+    player.playScene( getId(), 2, NO_DEFAULT_CAMERA, callback );
+  }
+
+  void Scene00003( Entity::Player& player )
+  {
+    auto callback = [ & ]( Entity::Player& player, const Event::SceneResult& result )
+    {
+    };
+
+    player.playScene( getId(), 3, NO_DEFAULT_CAMERA, callback );
+  }
+
+  void Scene00004( Entity::Player& player )
+  {
+    auto callback = [ & ]( Entity::Player& player, const Event::SceneResult& result )
+    {
+      if( result.param2 == 1 )
+      {
+        if( player.giveQuestRewards( getId(), 0 ) )
```

Comment:
Pass result.param3 instead of 0 to give player correct optional reward.

Reply:
Thank you!